### PR TITLE
Bugfix: Moved async where it should be

### PR DESCRIPTION
**Pull request by maudnals**
Thanks for the catch @beaufortfrancois!

### CODE REVIEW

File: product-review-suggestions/src/worker.js [hunk 1]
```diff
@@ -150,5 +149,10 @@ self.onmessage = function (message) {
       code: MESSAGE_CODE.RESPONSE_READY,
       payload: reviewHelperOutput,
     });
-  })();
+    return reviewHelperOutput;
+  } catch (error) {
+    console.error('[Worker] Error during inference:', error);
+    self.postMessage({ code: MESSAGE_CODE.INFERENCE_ERROR, payload: null });
+    return null;
```
Comment (beaufortfrancois) [inline product-review-suggestions/src/worker.js:156]: This is not required in this case.

File: product-review-suggestions/src/worker.js [hunk 2]
```diff
@@ -150,5 +149,10 @@ self.onmessage = function (message) {
       code: MESSAGE_CODE.RESPONSE_READY,
       payload: reviewHelperOutput,
     });
-  })();
+    return reviewHelperOutput;
```
Comment (beaufortfrancois) [inline product-review-suggestions/src/worker.js:152]: Same here. onmessage does not return anything.

File: perf-worker-gemma/src/worker.js [hunk 3]
```diff
@@ -26,10 +26,8 @@ self.onmessage = function (message) {
...
   self.postMessage({ code: MESSAGE_CODE.GENERATING_RESPONSE, payload: null });
 
-  (async function () {
-    // TODO handle errors (e.g. an inference error can happen when the input is too long). A simple try/catch isn't sufficient, we also need to terminate the previous/failing inference which I didn't figure out how to do
-    const response = await llmInference.generateResponse(message.data);
-    console.info('[Worker] Response generated');
-    self.postMessage({ code: MESSAGE_CODE.RESPONSE_READY, payload: response });
-  })();
+  // TODO handle errors (e.g. an inference error can happen when the input is too long). A simple try/catch isn't sufficient, we also need to terminate the previous/failing inference which I didn't figure out how to do
+  const response = await llmInference.generateResponse(message.data);
```
Comment (beaufortfrancois) [inline perf-worker-gemma/src/worker.js:30]: A try/catch there would be nice no?

File: product-review-suggestions/src/worker.js (hunk 1 above)
Comment (maudnals) [reply product-review-suggestions/src/worker.js]: Fixed

File: perf-worker-gemma/src/worker.js (hunk 3 above)
Comment (maudnals) [reply perf-worker-gemma/src/worker.js]: Yes! Added

File: perf-worker-gemma/src/worker.js [hunk 4]
```diff
@@ -26,10 +26,15 @@ self.onmessage = function (message) {
...
+  try {
+    response = await llmInference.generateResponse(message.data);
+  } catch (error) {
+    console.error('[Worker] Error during inference:', error);
+    self.postMessage({ code: MESSAGE_CODE.INFERENCE_ERROR, payload: null });
+    return;
+    // TODO Better handle errors (e.g. an inference error can happen when the input is too long). A simple try/catch isn't sufficient, we also need to terminate the previous/failing inference ("Previous invocation or loading is still ongoing.")
+  }
+  console.info('[Worker] Response generated');
+  self.postMessage({ code: MESSAGE_CODE.RESPONSE_READY, payload: response });
```
Comment (beaufortfrancois) [inline perf-worker-gemma/src/worker.js:39]:
```suggestion
  try {
    const response = await llmInference.generateResponse(message.data);
    console.info('[Worker] Response generated');
    self.postMessage({ code: MESSAGE_CODE.RESPONSE_READY, payload: response });
  } catch (error) {
    // TODO Better handle errors (e.g. an inference error can happen when the input is too long). A simple try/catch isn't sufficient, we also need to terminate the previous/failing inference ("Previous invocation or loading is still ongoing.")
    console.error('[Worker] Error during inference:', error);
    self.postMessage({ code: MESSAGE_CODE.INFERENCE_ERROR });
  }
  ```

**Review by beaufortfrancois — COMMENTED** [review]
LGTM with nit

**Review by beaufortfrancois — APPROVED** [review]

**Pull request merged**